drink/src/session/transcoding.rs

```diff
@@ -0,0 +1,23 @@
+use std::{collections::BTreeMap, rc::Rc};
+
+use contract_transcode::ContractMessageTranscoder;
+
+pub struct TranscoderRegistry<Contract: Ord> {
+    transcoders: BTreeMap<Contract, Rc<ContractMessageTranscoder>>,
+}
+
+impl<Contract: Ord> TranscoderRegistry<Contract> {
+    pub fn new() -> Self {
+        Self {
+            transcoders: BTreeMap::new(),
+        }
+    }
+
+    pub fn register(&mut self, contract: Contract, transcoder: &Rc<ContractMessageTranscoder>) {
+        self.transcoders.insert(contract, Rc::clone(transcoder));
+    }
+
+    pub fn get(&self, contract: &Contract) -> Option<Rc<ContractMessageTranscoder>> {
```

Comment:
Would a `remove` also make sense?

Reply:
actually I can't imagine a situation for that... even if you terminate a contract, there's no way you can deploy another contract with different metadata to the same address later